Sources/PackageModel/Destination.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+import Basic
+
+enum HostDestinationError: Swift.Error {
+    /// Couldn't find the Xcode installation.
+    case invalidInstallation(problem: String)
+}
+
+extension HostDestinationError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidInstallation(let problem):
+            return problem
+        }
+    }
+}
+
+enum DestinationFileError: Swift.Error {
+    /// Wrong version.
+    case destinationFileVersionMismatch(problem: String)
+}
+
+extension DestinationFileError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .destinationFileVersionMismatch(let problem):
+            return problem
+        }
+    }
+}
+
+public struct Destination {
```

Comment:
This looks like something which should be a part of toolchain and not each target and product.